integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py

```diff
@@ -298,24 +308,75 @@ def get_documents_by_id(
             documents.append(self.qdrant_to_haystack.point_to_document(record))
         return documents
 
-    def query_by_embedding(
+    def query_by_sparse(
         self,
-        query_embedding: List[float],
+        query_sparse_embedding: SparseEmbedding,
         filters: Optional[Dict[str, Any]] = None,
         top_k: int = 10,
         scale_score: bool = True,  # noqa: FBT001, FBT002
         return_embedding: bool = False,  # noqa: FBT001, FBT002
     ) -> List[Document]:
-        qdrant_filters = self.qdrant_filter_converter.convert(filters)
 
+        if not self.use_sparse_embeddings:
+            message = (
+                "Error: tried to query by sparse vector with a Qdrant "
+                "Document Store initialized with use_sparse_embeddings=False"
+            )
+            raise ValueError(message)
+
+        qdrant_filters = self.qdrant_filter_converter.convert(filters)
+        query_indices = query_sparse_embedding.indices
+        query_values = query_sparse_embedding.values
         points = self.client.search(
             collection_name=self.index,
-            query_vector=query_embedding,
+            query_vector=rest.NamedSparseVector(
+                name=SPARSE_VECTORS_NAME,
+                vector=rest.SparseVector(
+                    indices=query_indices,
+                    values=query_values,
+                ),
+            ),
             query_filter=qdrant_filters,
             limit=top_k,
             with_vectors=return_embedding,
         )
+        results = [self.qdrant_to_haystack.point_to_document(point) for point in points]
+        if scale_score:
+            for document in results:
+                score = document.score
+                score = float(1 / (1 + np.exp(-score / 100)))
+                document.score = score
+        return results
+
+    def query_by_embedding(
+        self,
+        query_embedding: List[float],
+        filters: Optional[Dict[str, Any]] = None,
+        top_k: int = 10,
+        scale_score: bool = True,  # noqa: FBT001, FBT002
+        return_embedding: bool = False,  # noqa: FBT001, FBT002
+    ) -> List[Document]:
+        qdrant_filters = self.qdrant_filter_converter.convert(filters)
 
+        if self.use_sparse_embeddings:
+            points = self.client.search(
+                collection_name=self.index,
+                query_vector=rest.NamedVector(
+                    name=DENSE_VECTORS_NAME,
+                    vector=query_embedding,
+                ),
+                query_filter=qdrant_filters,
+                limit=top_k,
+                with_vectors=return_embedding,
+            )
+        if not self.use_sparse_embeddings:
+            points = self.client.search(
+                collection_name=self.index,
+                query_vector=query_embedding,
+                query_filter=qdrant_filters,
+                limit=top_k,
+                with_vectors=return_embedding,
+            )
```

Comment:
Handle previous collection without named dense vector (and without sparse) so that there is not breaking change.